department-of-veterans-affairs/caseflow
# Appeals 4529 ensure vso instructional text appears on the convert to virtual form hearings application

Resolves [APPEALS-4529](https://vajira.max.gov/browse/APPEALS-4529)

### Description
On the Hearings Details page (`hearings/components/Details.jsx`) the cancel and submit button onClick methods are modified for all VSO users in the following fashions:
- `Cancel Button`: This will navigate a VSO user back to the Case Details page for the appeal the hearing is associated with.
- `Submit/"Convert Hearing to Virtual" Button`: This will also navigate a VSO to the Case Details page, and a success banner will appear on the page as visual confirmation that the hearing in question has been converted to virtual.

### Acceptance Criteria
The following text appears below the Convert to Virtual form header for VSO users

- [ ] Please work with the Veteran / Appellant to confirm that they have access to a phone or computer with internet access on the day of the virtual hearing. The Veteran / Appellant's email address will be used to send notifications for this hearing only.
- [ ] For VSO user Text appears instead of the existing verbiage: "Email notifications will be sent to the Appellant, POA / Representative, and Veterans Law Judge (VLJ)."
- [ ] Text defined in A/C 1.1 will not appear for non-VSO users

There is no change to the instructional text for non-VSO users
- [ ] the following instructional text continues to appear for non-VSO users: "Email notifications will be sent to the Appellant, POA / Representative, and Veterans Law Judge (VLJ)."

### Testing Plan
1. Open the Rails console, and grab a non-virtual hearing that has a disposition of `nil`:
```ruby
h = Hearing.where(disposition: nil).reject(&:virtual?).first
```

2. Get the UUID of the appeal the hearing is associated with:
```ruby
h.appeal.uuid
```

3. Log into Caseflow as a VSO user (e.g. Billie VSO)

4. Navigate to the Case Details for the appeal to load this page on top of the web history stack:
`/queue/appeals/<appeal-uuid>`

5. Navigate to the "Convert Hearing to Virtual" form in Caseflow Hearings for the hearing from step 1:
```ruby
h.uuid
```

`/hearings/<hearing-uuid>/details`

6. You should now be on the "Convert Hearing to Virtual" form. Check for the following instructional text to appear under the "Convert to Virtual" label at the top of the form: "Please work with the Veteran / Appellant to confirm that they have access to a phone or computer with internet access on the day of the virtual hearing. The Veteran / Appellant's email address will be used to send notifications for this hearing only."

7. Copy the URL so that it can be utilized later

8. Switch user to a Hearings Coordinator (e.g. BVASYELLOW)

9. Paste in the saved URL from step 7

10. This will land you on the Hearing Details Page.  In the Hearing Type dropdown, select virtual in order to go to the convert to virtual form

11. You should now be on the "Convert Hearing to Virtual" form. Check for the following instructional text to appear under the "Convert to Virtual" label at the top of the form: "Email notifications will be sent to the Veteran, POA / Representative, and Veterans Law Judge (VLJ)."
